src/spec/Paths.php

```diff
@@ -39,16 +39,24 @@ class Paths implements SpecObjectInterface, ArrayAccess, Countable, IteratorAggr
 
     /**
      * Create an object from spec data.
-     * @param array $data spec data read from YAML or JSON
+     * @param PathItem[]|array[] $data spec data read from YAML or JSON
      * @throws TypeErrorException in case invalid data is supplied.
      */
     public function __construct(array $data)
     {
         foreach ($data as $path => $object) {
             if ($object === null) {
                 $this->_paths[$path] = null;
-            } else {
+            } elseif (is_array($object)) {
                 $this->_paths[$path] = new PathItem($object);
+            } elseif ($object instanceof PathItem) {
+                $this->_paths[$path] = $object;
+            } else {
+                $givenType = gettype($object);
+                if ($givenType === 'object') {
+                    $givenType = get_class($object);
+                }
+                throw new \TypeError(sprintf('Path MUST be either array or PathItem object, "%s" given', $givenType));
```

Comment:
this should throw `TypeErrorException` as annotated in phpdoc

Reply:
also instantiation logic in Paths and Responses is very similar but we have the type check with exception here but not in Responses. Imo it should follow same rules.

Reply:
we should probably add it in all three places. i.e. if an object is given, verify that it is correct object as expected in this place.